includes/payment-methods/class-wc-stripe-upe-payment-method.php

```diff
@@ -543,16 +543,20 @@ public function payment_fields() {
 				<input type="hidden" class="wc-stripe-is-deferred-intent" name="wc-stripe-is-deferred-intent" value="1" />
 			</fieldset>
 			<?php
+
 			if ( $this->should_show_save_option() ) {
 				$force_save_payment = ( $display_tokenization && ! apply_filters( 'wc_stripe_display_save_payment_method_checkbox', $display_tokenization ) ) || is_add_payment_method_page();
 				if ( is_user_logged_in() ) {
 					$this->save_payment_method_checkbox( $force_save_payment );
 				}
 			}
+
 			if ( $display_tokenization ) {
 				$this->tokenization_script();
 				$this->saved_payment_methods();
 			}
+
+			do_action( 'wc_stripe_payment_fields_' . $this->id, $this->id );
```

Comment:
This action existed on the legacy checkout ([here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/issue/3453-add-update-all-subscription-pm-checkbox/includes/class-wc-gateway-stripe.php#L267)), but it was never triggered from the UPE versions of the `payment_fields()`. 

I think it's fine to reintroduce it, however, happy to be persuaded otherwise.

Reply:
What's the alternative?

If I understand correctly, the alternative would be to introduce a new `do_action` hook name that's used for UPE specifcally and then attach `display_update_subs_payment_checkout()` to this new hook?

I think it's fine to use the same hook for legacy and UPE :)